File: programs/libc/src/unistd/getuid.c
#include <signal.h>
#include <user_syscall.h>

/// The getuid() function shall return the real user ID of the calling
/// process. The getuid() function shall not modify errno.

uid_t getuid(void)
{
	/*
	 * This function is always successful.
	 */
	uid_t ret = _user_syscall(GETUID, 0);
	return ret;
}

File: rust_kernel/src/taskmaster/syscall/getuid.rs
//! sys_getuid()

use super::scheduler::SCHEDULER;
use super::SysResult;

/// The getuid() function shall return the real user ID of the calling
/// process. The getuid() function shall not modify errno.
pub fn sys_getuid() -> SysResult<u32> {
    Ok(unpreemptible_context!({
        SCHEDULER.lock().current_thread_group().credentials.uid as u32
    }))
}
